lib/spreedly/base.ex

```diff
@@ -99,6 +99,12 @@ defmodule Spreedly.Base do
   end
 
   @spec headers(Environment.t) :: headers
+  defp headers(env) when is_nil(env) do
```

Comment:
One of the super cool things about Elixir is that we can do neat things with function head matches. We can rewrite this as 

```
 defp headers(nil), do: [{"Content-Type", "application/json"}]
```

If we wanted to avoid code duplication, we could break the content type to its own function.

```
defp headers(nil), do: [content_type()]
defp headers(env) do
...
end

def content_type, do: {"Content-Type", "application/json"}
```

Reply:
That makes the code so much neater! Also, I didn't add documentation. Will make those changes.